docs/api/locking.md

```diff
@@ -0,0 +1,449 @@
+// Content-Type: application/vnd.git-lfs+json
+{
+  "message": "You must have push access to create a lock",
+  "documentation_url": "https://lfs-server.com/docs/errors",
+  "request_id": "123"
+}
+```
+
+### Error Response
+
+* `message` - String error message.
+* `request_id` - Optional String unique identifier for the request. Useful for
+debugging.
+* `documentation_url` - Optional String to give the user a place to report
+errors.
+
+```js
+// HTTP/1.1 500 Internal server error
+// Content-Type: application/vnd.git-lfs+json
+{
+  "message": "already created lock",
+  "documentation_url": "https://lfs-server.com/docs/errors",
+  "request_id": "123"
+}
+```
+
+## List Locks
+
+The client can request the current active locks for a repository by sending a
+`GET` to `/locks` (appended to the LFS server url, as described above). The
+properties are sent as URI query values, instead of through a JSON body:
+
+* `path` - Optional string path to match against locks on the server.
+* `id` - Optional string ID to match against a lock on the server.
+* `cursor` - The optional string value to continue listing locks. This value
+should be the `next_cursor` from a previous request.
+* `limit` - The integer limit of the number of locks to return. The server
+should have its own upper and lower bounds on the supported limits.
+
+```js
+// GET https://lfs-server.com/locks?path=&id=&cursor=&limit=
+// Accept: application/vnd.git-lfs+json
+// Authorization: Basic ... (if needed)
+```
+
+### Successful Response
+
+A successful response will list the matching locks:
+
+* `locks` - Array of matching Lock objects. See the "Create Lock" successful
+response section to see what Lock properties are possible.
+* `next_cursor` - Optional string cursor that the server can return if there
+are more locks matching the given filters. The client will re-do the request,
+setting the `?cursor` query value with this `next_cursor` value.
+
+Note: If the server has no locks, it must return an empty `locks` array.
+
+```js
+// HTTP/1.1 200 Ok
+// Content-Type: application/vnd.git-lfs+json
+{
+  "locks": [
+    {
+      "id": "some-uuid",
+      "path": "/path/to/file",
+      "locked_at": "2016-05-17T15:49:06+00:00",
+      "committer": {
+        "name": "Jane Doe"
+      }
+    }
+  ],
+  "next_cursor": "optional next ID",
+}
+```
+
+### Unauthorized Response
+
+Lock servers should require that users have pull access to the repository before
+they can list locks.
+
+* `message` - String error message.
+* `request_id` - Optional String unique identifier for the request. Useful for
+debugging.
+* `documentation_url` - Optional String to give the user a place to report
+errors.
+
+```js
+// HTTP/1.1 403 Forbidden
+// Content-Type: application/vnd.git-lfs+json
+{
+  "message": "You must have pull access to list locks",
+  "documentation_url": "https://lfs-server.com/docs/errors",
+  "request_id": "123"
+}
+```
+
+### Error Response
+
+* `message` - String error message.
```

Comment:
This may be out of scope of the current PR, but we could probably format a lot of these request/response fields lists with a table:

| Name | Type | Description |
| --- | ---- | --- |
| `message` | String | Error message. |